### ISSUE TITLE
 list index out of range when running telepresence

### ISSUE THREAD

**Issue by Krever** [telepresenceio/telepresence#1299]
### What were you trying to do?

run `telepresence`

### What did you expect to happen?

dunno, just following quickstart

### What happened instead?

Got the error attached below

### Automatically included information

Command line: `['/usr/local/bin/telepresence']`
Version: `0.104`
Python version: `3.7.7 (default, Mar 10 2020, 15:43:33) 
[Clang 11.0.0 (clang-1100.0.33.17)]`
kubectl version: `Client Version: v1.18.0 // Server Version: v1.17.4`
oc version: `(error: [Errno 2] No such file or directory: 'oc': 'oc')`
OS: `Darwin wp-xps-2.home 19.4.0 Darwin Kernel Version 19.4.0: Wed Mar  4 22:28:40 PST 2020; root:xnu-6153.101.6~15/RELEASE_X86_64 x86_64`

```
Traceback (most recent call last):
  File "/usr/local/bin/telepresence/telepresence/cli.py", line 135, in crash_reporting
    yield
  File "/usr/local/bin/telepresence/telepresence/main.py", line 65, in main
    remote_info = start_proxy(runner)
  File "/usr/local/bin/telepresence/telepresence/proxy/__init__.py", line 138, in start_proxy
    run_id=run_id,
  File "/usr/local/bin/telepresence/telepresence/proxy/remote.py", line 144, in get_remote_info
    runner, deployment_name, deployment_type, run_id=run_id
  File "/usr/local/bin/telepresence/telepresence/proxy/remote.py", line 91, in get_deployment_json
    return json.loads(output)["items"][0]
IndexError: list index out of range

```

Logs:

```
 STG --namespace crow delete --ignore-not-found svc,deploy --selector=telepresence=83f2b5dd5f024479912c7a72a25a3496
   2.8   9 | No resources found
   2.8 TEL | [9] ran in 0.32 secs.
   2.8 TEL | [10] Running: kubectl --context STG --namespace crow run --restart=Always --limits=cpu=1000m,memory=256Mi --requests=cpu=25m,memory=64Mi telepresence-1586343097-568764-66395 --image=datawire/telepresence-k8s:0.104 --labels=telepresence=83f2b5dd5f024479912c7a72a25a3496
   3.1  10 | pod/telepresence-1586343097-568764-66395 created
   3.1 TEL | [10] ran in 0.32 secs.
   3.1 TEL | END SPAN deployment.py:108(create_new_deployment)    0.6s
   3.1 TEL | BEGIN SPAN remote.py:142(get_remote_info)
   3.1 TEL | BEGIN SPAN remote.py:75(get_deployment_json)
   3.1 TEL | [11] Capturing: kubectl --context STG --namespace crow get deployment -o json --selector=telepresence=83f2b5dd5f024479912c7a72a25a3496
   3.4 TEL | [11] captured in 0.25 secs.
   3.4 TEL | END SPAN remote.py:75(get_deployment_json)    0.2s

```

**Comment by marun** [comment]
@Krever Can you provide the command line that resulted in this output?

Note that you're using a 1.18 version of kubectl, which includes a change in behavior to `kubectl run`: 

https://github.com/telepresenceio/telepresence/issues/1297

tl;dr Running `kubectl run` and then attempting to target telepresence will fail due to a pod being created but telepresence requiring a deployment.

**Comment by jtyers** [comment]
see workaround in #1297

**Comment by Krever** [comment]
Hey @marun , the command I executed was just plain `telepresence` without any arguments or options. This is what quickstart on the main page advertises AFAIR.

**Comment by marun** [comment]
@Krever For a bare invocation, `telepresence` calls `kubectl run` with the assumption that it will create a deployment:

https://github.com/telepresenceio/telepresence/blob/master/telepresence/proxy/deployment.py#L130

As per my previous comment, `kubectl run` breaks that assumption as of version 1.18. @jtyers [suggests possible workarounds](https://github.com/telepresenceio/telepresence/issues/1297#issuecomment-611404863) on #1297.

**Comment by Krever** [comment]
I understand now, thanks for the explanation. Should we close the issue? Anyway, I created it thanks to the automated message, so if it's a known issue/limitation than this message might be adjusted.

**Comment by AndreasPresthammer** [comment]
Also experienced this issue. Snap had upgrade kubectl to 1.18 which broke this for me. I resolved it by downgrading to 1.14 (which match my cluster version).

Snap downgrade:
sudo snap refresh kubectl --channel=1.14/stable --classic

**Comment by marun** [comment]
> I understand now, thanks for the explanation. Should we close the issue? Anyway, I created it thanks to the automated message, so if it's a known issue/limitation than this message might be adjusted.

I'd suggest leaving it up for now, the presence of many issues on the topic increases the perceived severity of the issue. I've proposed a fix btw (#1312) so hopefully the breakage won't last too much longer.

**Comment by ark3** [comment]
This was fixed in release 0.105.
